### PR TITLE
[mkbundle] Add support for generating monodroid compatible mkbundle code at runtime

### PR DESCRIPTION
**Pull request by dellis1972**
monodroid used to build its own version of mkbundle since it
requires slightly different code than the desktop. This
logic was protected in a #if block. Now that monodroid is
trying to make use of the shipped mkbundle rather than use
its own we need to turn this logic on at runtime.

A new parameter has been added

	--bundled-header

When this option is passed, the code which skips including

	mono/metadata/mono-config.h

will be skipped in favour of defining the required structures
and entry points directly.

### CODE REVIEW

**Comment by akoeplinger** [comment]
@alexischr could you please take a look?

I also wonder if we could make this version the default, why do we include mono header files here?

File: mcs/tools/mkbundle/mkbundle.cs [hunk 1]
```diff
@@ -1238,6 +1242,7 @@ static void Help ()
 				   "    --skip-scan          Skip scanning assemblies that could not be loaded (but still embed them).\n" +
 				   "    --i18n ENCODING      none, all or comma separated list of CJK, MidWest, Other, Rare, West.\n" +
 				   "    -v                   Verbose output\n" + 
+				   "    --profile P          produce code for the target profile `P'\n" +
```
Comment (akoeplinger) [inline mcs/tools/mkbundle/mkbundle.cs]: could you please update this to --bundled-header?

**Comment by kumpera** [comment]
This change is much appreciated, thanks @dellis1972!

**Comment by migueldeicaza** [comment]
Dean, question, is there a reason that you can not use the new-style mkbundle for Android?

**Comment by dellis1972** [comment]
@migueldeicaza to be honest I have not tried it. I will investigate to see if it will work, I'll read up on the new mkbundle. 
We should probably still merge this PR since its the quickest and easiest option at the moment.

**Comment by dellis1972** [comment]
@akoeplinger should I change this to ` --no-mono-include`?

**Pull request merged**